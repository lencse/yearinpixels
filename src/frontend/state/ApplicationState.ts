import * as dateformat from 'dateformat'
import User from '../auth/User'
import NullUser from '../auth/NullUser'
import DayStore from '../calendar/DayStore'

export default interface ApplicationState {

    user: User

    days: DayStore,

    currentDate: Date

}

export function initial(): ApplicationState {
    return {
        user: new NullUser(),
        days: new DayStore([
            {
                date: new Date('2019-01-01'),
                mood: 1,
                comment: ''
            },
            {
                date: new Date('2019-01-02'),
                mood: 1,
                comment: ''
            },
            {
                date: new Date('2019-01-03'),
                mood: 2,
                comment: ''
            },
            {
                date: new Date('2019-01-04'),
                mood: 5,
                comment: ''
            },
            {
                date: new Date('2019-03-15'),
                mood: 6,
                comment: ''
            }
        ]),
        currentDate: new Date(dateformat(new Date(), 'yyyy-mm-dd'))
    }
}
