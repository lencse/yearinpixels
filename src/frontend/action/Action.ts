import ApplicationState from '../state/ApplicationState'

export default interface Action {

    transform(state: ApplicationState): ApplicationState

}
