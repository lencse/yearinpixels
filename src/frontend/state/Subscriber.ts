import ApplicationState from './ApplicationState'

export default interface Subscriber {

    init(state: ApplicationState): void

    update(state: ApplicationState): void

}
