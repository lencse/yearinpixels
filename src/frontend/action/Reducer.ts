import Action from './Action'

export default interface Reducer {

    apply(action: Action): void

}
