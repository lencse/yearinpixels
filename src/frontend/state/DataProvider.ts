import Subscriber from './Subscriber'

export default interface DataProvider {

    addSubscriber(subscriber: Subscriber): void

}
