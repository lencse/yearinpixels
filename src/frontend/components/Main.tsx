import * as React from 'react'
import DataProvider from '../state/DataProvider'
import ApplicationState from '../state/ApplicationState'
import Subscriber from '../state/Subscriber'
import Table from './Table'
import Reducer from '../action/Reducer'
import Panel from './Panel'

interface MainProps {

    dataProvider: DataProvider

    reducer: Reducer

}

export default class Main extends React.Component<MainProps, ApplicationState> implements Subscriber {

    constructor(props: MainProps, context) {
        super(props, context)
        props.dataProvider.addSubscriber(this)
    }

    public render(): React.ReactNode {
        return (
            <div>
                <Table data={{...this.state}} reducer={ this.props.reducer } />
                <Panel data={{...this.state}} reducer={ this.props.reducer } />
            </div>
        )
    }

    public init(initial: ApplicationState): void {
        this.state = initial
    }

    public update(state: ApplicationState): void {
        this.setState(state)
    }

}
