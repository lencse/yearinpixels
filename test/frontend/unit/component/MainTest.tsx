import * as React from 'react'
import * as renderer from 'react-test-renderer'
import Main from '../../../../src/frontend/components/Main'
import Reducer from '../../../../src/frontend/action/Reducer'
import Action from '../../../../src/frontend/action/Action'
import DataProvider from '../../../../src/frontend/state/DataProvider'
import Subscriber from '../../../../src/frontend/state/Subscriber'
import { initial } from '../../../../src/frontend/state/ApplicationState'

describe('Main', () => {
    it('Main component renders', () => {
        const reducer: Reducer = {
            apply(action: Action) {
            }
        }
        const dataProvider: DataProvider = {
            addSubscriber(subscriber: Subscriber) {
                subscriber.init(initial())
            }
        }
        const main = renderer.create(
            <Main reducer={ reducer } dataProvider={ dataProvider} />
        ).toJSON()
        expect(main).toMatchSnapshot()
    })
})
