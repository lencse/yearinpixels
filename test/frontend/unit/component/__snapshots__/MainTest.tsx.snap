// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Main component renders 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <th />
        <th>
          J
        </th>
        <th>
          F
        </th>
        <th>
          M
        </th>
        <th>
          A
        </th>
        <th>
          M
        </th>
        <th>
          J
        </th>
        <th>
          J
        </th>
        <th>
          A
        </th>
        <th>
          S
        </th>
        <th>
          O
        </th>
        <th>
          N
        </th>
        <th>
          D
        </th>
      </tr>
      <tr>
        <th>
          1
          .
        </th>
        <td
          className="cell--real cell--mood-1"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          2
          .
        </th>
        <td
          className="cell--real cell--mood-1"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          3
          .
        </th>
        <td
          className="cell--real cell--mood-2"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          4
          .
        </th>
        <td
          className="cell--real cell--mood-5"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          5
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          6
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          7
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          8
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          9
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          10
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          11
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          12
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          13
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          14
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          15
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-6"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          16
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          17
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          18
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          19
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          20
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          21
          .
        </th>
        <td
          className="cell--real cell--mood-0 cell--current"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          22
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          23
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          24
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          25
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          26
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          27
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          28
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          29
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--fake"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          30
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--fake"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
      <tr>
        <th>
          31
          .
        </th>
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--fake"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--fake"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--fake"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--fake"
        />
        <td
          className="cell--real cell--mood-0"
        />
        <td
          className="cell--fake"
        />
        <td
          className="cell--real cell--mood-0"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
